File: Exercícios/Capitulo_2/Propostos/q6.cpp
#include <iostream>
#include <tgmath.h>
#include <math.h>
using namespace std;

void calcular_raizes(int a, int b, int delta)
{
    if (delta < 0)
    {
        cout << "\nNão possui raizes reais.";
    }
    else if (delta > 0)
    {
        cout << "\nPossui duas raizes reais: \n";
        int x1 = (-b + sqrt(delta)) / (2 * a), x2 = (-b - sqrt(delta)) / (2 * a);

        cout << "\nx1 = " << x1;
        cout<< "\nx2 = " << x2;
    }
    else
    {
        cout << "\nPossui apenas uma raiz real: ";
        int x = (-b + sqrt(delta)) / (2 * a);

        cout << "\nx = " << x;
    }
}

int calcular_delta(int a, int b, int c)
{
    int delta = pow(b, 2) - (4 * a * c);

    return delta;
}

void input(int &a, int &b, int &c)
{
    cout << "Valor de a = ";
    cin >> a;
    cout << "Valor de b = ";
    cin >> b;
    cout << "Valor de c = ";
    cin >> c;
}

int main()
{
    int a, b, c, delta;

    input(a, b, c);
    delta = calcular_delta(a, b, c);
    calcular_raizes(a, b, delta);

    return 0;
}

// a*pow(x, 2) + b*x + c = 0

File: Exercícios/Capitulo_2/Propostos/q6.py
from math import sqrt

# a*pow(x, 2) + b*x + c = 0

""" Constantes """
a = int(input("Valor de a = "))
b = int(input("Valor de b = "))
c = int(input("Valor de c = "))

delta = pow(b, 2) - (4 * a * c)
""" Se delta > 0, então há x1 e x2
    Se delta = 0, então há apenas x1
    Se delta < 0, então não há raizes reais"""

if delta < 0:
    print("Não há raizes reais.")

elif delta > 0:
    x1 = (-b + sqrt(delta)) // (2 * a)
    x2 = (-b - sqrt(delta)) // (2 * a)

    print("Possui duas raízes reais: ")
    print(f"x1 = {x1}", f"x2 = {x2}")

else:  # delta == 0
    x = (-b + sqrt(delta)) // (2 * a)
    
    print("Possui apenas uma raíz real: ")
    print(f"x = {x}")
